src/openforms/restapis/models.py

```diff
@@ -0,0 +1,82 @@
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+import requests
+from djchoices.choices import ChoiceItem, DjangoChoices
+
+from .compat import jwt_encode
+
+JWT_ALG = "HS256"
+
+
+class AuthTypes(DjangoChoices):
+    no_auth = ChoiceItem("no_auth", _("No authentication"))
+    basic_auth = ChoiceItem("basic_auth", _("Basic authentication"))
+    api_key = ChoiceItem("api_key", _("API key"))
+    jwt = ChoiceItem("jwt", _("JWT"))
+    custom_header = ChoiceItem("custom_header", _("Custom header"))
+
+
+class RestAPI(models.Model):
```

Comment:
I propose dropping this, as we keep spreading the various service configurations across different DB tables, and thus in the admin interface.

Maybe you could make this model a proxy model of `zgw_consumers.Service` to build a client based on requests without any OAS stuff, and extract the auth parameters/credentials as headers from the `zgw_consumers.Service` items.

Not sure why the JWT options are here, are they relevant? It strikes me as odd to have the payload defined in the DB while it contains dynamic claims like "issued at" timestamps and expiry timestamps.

Reply:
Dropped in favor of 10-line client.